### DISCUSSION TITLE
Issues with Combiner Aliases

### DISCUSSION THREAD

**Discussion by RoKrypto** [wintercms/winter#622] [Q&A]
Greetings. I've experienced issues with the following combiner:

```html
<script defer src="{{ ['@jquery', '@framework.extras.js'] | theme }}"></script>
```

It doesn't print the pair of scripts you'd expect. After some tests, I found out the only way it works is by splitting them:

```html
<script defer src="{{ ['@jquery'] | theme }}"></script>
<script defer src="{{ ['@framework.extras.js'] | theme }}"></script>
```

In case no aliases are used, the combiner works perfectly 👍

In addition, the following alias doesn't work either and throws an exception that said the source file doesn't exist:

```html
<link rel="stylesheet" href="{{ ['@framework.extras.css'] | theme }}">
```

Suggestions on how to tackle these issues will be much appreciated.

**Comment by LukeTowers** [comment]
@RoKrypto use `@jquery`, `@framework`, & `@framework.extras` together

**Reply by RoKrypto** [reply]
Hi @LukeTowers . It works. Thank you for your support. Nevertheless, I still can't get the CSS printed.

```html
<link rel="stylesheet" href="{{ ['@framework.extras.css'] | theme }}">
```

**Reply by arvislacis** [reply]
I am not 100% sure if I am right but I think that there is no `@framework.extras.css` for CSS, it's either `@framework.extras` or `@framework` or..., probably Luke will know better.

**Reply by RoKrypto** [reply]
According to the documentation, there's a CSS and a JS file related to the framework that helps working with AJAX, especially with the data attributes API.

**Comment by arvislacis** [comment]
If you need to included these JS and CSS files in the most of pages and use combiner, a different approach which I use and like personally is to use `$this->addCss([ ... ]);` and `$this->addJs([ ... ]);` in `function onStart()` in layout file and then simply use `{% styles %}` and `{% scripts %}` in HTML part of layout file.

```
description = "Default layout"
==
<?php

function onStart() {
    $this->addCss(['assets/css/style-reset.css', 'assets/css/style.css',]);
    $this->addJs(['assets/js/jquery.min.js', '@framework', 'assets/js/script.js']);
}
==
<!DOCTYPE html>
<html>
<head>
    [..]
    {% styles %}
</head>
<body>
    [..]
    {% scripts %}
</body>
</html>
```

**Reply by RoKrypto** [reply]
I've got everything in place in terms of CSS and JS for my project, but the only one missing is the CSS file from the framework. I need to work with its alias instead of its Twig tag to group the files the way I need it.
